SPECS/picosat/picosat.spec

```diff
@@ -0,0 +1,241 @@
+The library has a simple API which is similar to that of previous
+solvers by the same authors.  This version of the library is built for
+use with R projects.
+
+%package libs
+Summary:        A SAT solver library
+
+%description libs
+The PicoSAT library, which contains routines that solve the SAT problem.
+The library has a simple API which is similar to that of previous
+solvers by the same authors.
+
+%package devel
+Summary:        Development files for PicoSAT
+Requires:       %{name}-R%{?_isa} = %{version}-%{release}
+Requires:       %{name}-libs%{?_isa} = %{version}-%{release}
+
+%description devel
+Headers and other development files for PicoSAT.
+
+%prep
+%setup -q
+%patch0
+
+%build
+# The configure script is NOT autoconf-generated and chooses its own CFLAGS,
+# so we mimic its effects instead of using it.
+
+# Build the version with R support
+sed -e "s/@CC@/gcc/" \
+    -e "s|@CFLAGS@|$RPM_OPT_FLAGS -D_GNU_SOURCE=1 -DNDEBUG -DRCODE -I/usr/lib64/R/include|" \
+    -e "s|-Xlinker libpicosat.so|-Xlinker libpicosat.so.0 $RPM_LD_FLAGS -L/usr/lib64/R/lib |" \
+    -e "s/libpicosat/libpicosat-R/g" \
+    -e "s/-lpicosat/-lpicosat-R/g" \
+    -e "s/@TARGETS@/libpicosat-R.so/" \
+  makefile.in > makefile
+%make_build
+
+# Build the version with trace support
+sed -e "s/@CC@/gcc/" \
+    -e "s|@CFLAGS@|$RPM_OPT_FLAGS -D_GNU_SOURCE=1 -DNDEBUG -DTRACE|" \
+    -e "s|-Xlinker libpicosat.so|-Xlinker libpicosat.so.0 $RPM_LD_FLAGS|" \
+    -e "s/libpicosat/libpicosat-trace/g" \
+    -e "s/-lpicosat/-lpicosat-trace/g" \
+    -e "s/@TARGETS@/libpicosat-trace.so picosat picomus/" \
+  makefile.in > makefile
+%make_build
+mv picosat picosat.trace
+
+# Build the fast version.
+# Note that picomus needs trace support, so we don't rebuild it.
+rm -f *.o *.s config.h
+sed -e "s/@CC@/gcc/" \
+    -e "s|@CFLAGS@|$RPM_OPT_FLAGS -D_GNU_SOURCE=1 -DNDEBUG|" \
+    -e "s|-Xlinker libpicosat.so|-Xlinker libpicosat.so.0 $RPM_LD_FLAGS|" \
+    -e "s/@TARGETS@/libpicosat.so picosat picomcs picogcnf/" \
+  makefile.in > makefile
+%make_build
+
+%install
+# Install the header file
+mkdir -p $RPM_BUILD_ROOT%{_includedir}
+cp -p picosat.h $RPM_BUILD_ROOT%{_includedir}
+
+# Install the libraries
+mkdir -p $RPM_BUILD_ROOT%{_libdir}
+install -m 0755 -p libpicosat-R.so \
+  $RPM_BUILD_ROOT%{_libdir}/libpicosat-R.so.0.0.%{version}
+ln -s libpicosat-R.so.0.0.%{version} $RPM_BUILD_ROOT%{_libdir}/libpicosat-R.so.0
+ln -s libpicosat-R.so.0 $RPM_BUILD_ROOT%{_libdir}/libpicosat-R.so
+install -m 0755 -p libpicosat-trace.so \
+  $RPM_BUILD_ROOT%{_libdir}/libpicosat-trace.so.0.0.%{version}
+ln -s libpicosat-trace.so.0.0.%{version} $RPM_BUILD_ROOT%{_libdir}/libpicosat-trace.so.0
+ln -s libpicosat-trace.so.0 $RPM_BUILD_ROOT%{_libdir}/libpicosat-trace.so
+install -m 0755 -p libpicosat.so \
+  $RPM_BUILD_ROOT%{_libdir}/libpicosat.so.0.0.%{version}
+ln -s libpicosat.so.0.0.%{version} $RPM_BUILD_ROOT%{_libdir}/libpicosat.so.0
+ln -s libpicosat.so.0 $RPM_BUILD_ROOT%{_libdir}/libpicosat.so
```

Comment:
>$RPM_BUILD_ROOT [](http://example.com/codeflow?start=22&length=15)

Nit: I believe we've more or less standardized on using %{buildroot} instead of $RPM_BUILD_ROOT. They should effectively be the same, though.